core/metacling/src/TCling.cxx

```diff
@@ -5354,12 +5354,15 @@ Int_t TCling::LoadLibraryMap(const char* rootmapfile)
 
    // Process the forward declarations collected
    cling::Transaction* T = nullptr;
-   auto compRes= fInterpreter->declare(uniqueString.Data(), &T);
-   assert(cling::Interpreter::kSuccess == compRes && "A declaration in a rootmap could not be compiled");
 
-   if (compRes!=cling::Interpreter::kSuccess){
-      Warning("LoadLibraryMap",
+   if (!fCxxModulesEnabled) {
```

Comment:
Isn't the correct test: 'if module-is-enabled && this-library-has-a-pcm' ?

Reply:
That's the challenge. We do not have that information at that point. Worse, even we have a pcm, the loading of the pcm can fail.

Reply:
Well, it seems it is a challenge we must tackled.  *If* I am reading this correctly, this PR means that autoloading is disabled for library/dictionary that do not use pcm but module support is enabled (i.e. for all intent and purposes that would mean that we do **not** support the mixed case)

Reply:
It seems we can check here if we have a pcm with the expected name on disk. We should change TCling::RegisterModule such that if we fail to load the nominated pcm, we should alternatively load the corresponding rootmap. We are somewhat safe with that solution as the current implementation preloads all available pcm files and virtually LoadLibraryMap happens at the same time as LoadModule.

Reply:
requiresRootMap(rootmapfile, fInterpreter) is essentially doing the same thing as 'if module-is-enabled && this-library-has-a-pcm', but please give me some time to try those out ( do not review yet label ;) )

Reply:
Now we seem to have is interface as per `HasPCMForLibrary`. We should use it here.